File: velolabs_repos/lattis_android/app/src/main/java/com/lattis/ellipse/bluetooth/HardwareStateMapper.java
package com.lattis.ellipse.bluetooth;

import androidx.annotation.NonNull;

import com.lattis.ellipse.data.network.base.AbstractDataMapper;
import com.lattis.ellipse.domain.model.Lock;

import javax.inject.Inject;

import io.lattis.ellipse.sdk.Ellipse.Hardware.State;


public class HardwareStateMapper extends AbstractDataMapper<State, Lock.Hardware.State> {

    private LockPositionMapper lockPositionMapper;

    @Inject
    public HardwareStateMapper(LockPositionMapper lockPositionMapper) {
        this.lockPositionMapper = lockPositionMapper;
    }

    @NonNull
    @Override
    public Lock.Hardware.State mapIn(@NonNull State state) {
        return new Lock.Hardware.State(lockPositionMapper.mapOut(state.getPosition()),
                state.getBatteryLevel(),
                state.getRssiLevel(),
                state.getTemperature());
    }

    @NonNull
    @Override
    public State mapOut(@NonNull Lock.Hardware.State state) {
        return null;
    }
}

File: velolabs_repos/lattis_android_v2/data/src/main/java/com/lattis/data/mapper/bluetooth/HardwareStateMapper.kt
package com.lattis.data.mapper.bluetooth

import com.lattis.data.mapper.AbstractDataMapper
import com.lattis.domain.models.Lock
import io.lattis.ellipse.sdk.Ellipse
import javax.inject.Inject

class HardwareStateMapper @Inject constructor(private val lockPositionMapper: LockPositionMapper) :
    AbstractDataMapper<Ellipse.Hardware.State?, Lock.Hardware.State?>() {
    override fun mapIn(state: Ellipse.Hardware.State?): Lock.Hardware.State {
        return Lock.Hardware.State(
            lockPositionMapper.mapOut(state!!.position),
            state?.batteryLevel,
            state?.rssiLevel,
            state?.temperature
        )
    }

    override fun mapOut(state: Lock.Hardware.State?): Ellipse.Hardware.State {
        return Ellipse.Hardware.State(lockPositionMapper.mapIn(state?.position),state?.batteryLevel!!,state.rssiLevel!!,state.temperature!!)
    }

}